src/task/detect-bad-words-task.ts

```diff
@@ -41,19 +41,32 @@ export class DetectBadWordsTask extends BaseValidationTask<string[]> {
     this.status = 'RUNNING';
 
     const { itemId } = this.input;
+    const { id: processId } = await this.validationService.getProcessId('bad-words-detection', handler);
+
+    // Add record of this validation process
+    const itemValidationEntry = await this.validationService.createItemValidation(itemId, processId, handler);
+    console.log(itemValidationEntry);
+
     const item = await this.itemService.get(itemId, handler);
-    const suspiciousFields = this.checkBadWrods([
+    const suspiciousFields = this.checkBadWords([
       {name: 'name', value: item.name}, 
       {name: 'description', value: stripHtml(item.description)}
     ]);
-    if (suspiciousFields.length > 0) {
-      this.status = 'FAIL';
-      this._result = suspiciousFields;
+
+    // Update record after the process finishes
+    const updatedItemValidationEntry = suspiciousFields.length > 0 ? 
+      {...itemValidationEntry, status: Status.Failure, result: suspiciousFields.toString()} :
+      {...itemValidationEntry, status: Status.Success, result: ''};
+    await this.validationService.updateItemValidation(updatedItemValidationEntry as ItemValidation, handler);
+
+    // set task status, result and message
+    this.status = 'OK';  // The task status is always 'OK', since the task itself completed successfully
```

Comment:
For security I would set the status after you finish everything (so after the if-else clauses).